pgmpy/factors/hybrid/FunctionalCPD.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+
+from pgmpy.factors.base import BaseFactor
+
+
+class FunctionalCPD(BaseFactor):
+    def __init__(self, variable, fn, parents=None):
+        """
+        Parameters:
+        ----------
+        variable: str
+            Name of the variable for which this CPD is defined.
+        fn: callable
+            A function that takes a dictionary of parent variable values
+            and returns a sampled value for the variable.
+        parents: list[str], optional
+            List of parent variable names (default is None for no parents).
+
+        Examples
+        --------
+        # For P(Y| X1, X2) = N(0.2x1 + 0.3x2 + 1.0; 1)
+
+        >>> cpd = FunctionalCPD(
+            variable="x3",
+            fn=lambda parent_sample: np.random.normal(
+                0.2 * parent_sample["x1"] + 0.3 * parent_sample["x2"] + 1.0, 1
+            ),
+            parents=["x1", "x2"],
+        )
+        >>> cpd.variable
+        'x3'
+        >>> cpd.parents
+        ['x1', 'x2']
+        """
+        self.variable = variable
+        if not callable(fn):
+            raise ValueError("`fn` must be a callable function.")
+        self.fn = fn
+        self.parents = parents if parents else []
+
+    def sample(self, parent_sample=None):
```

Comment:
I would change this function to accept a `n_samples` argument and accept `parent_sample` as a dataframe object. In the broader context, the user would have a Bayesian Network with FunctionalCPD associated with each of the nodes. They would then sample from this network, which would translate into calling this `sample` function for each of these FunctionalCPDs.